java/client/src/org/openqa/selenium/remote/RemoteWebDriver.java

```diff
@@ -331,6 +332,11 @@ public WebElement findElement(By locator) {
     }
   }
 
+  public String printPage(PrintOptions options) {
```

Comment:
If this is something we expect all `WebDriver` instances to support (and we do, since it's in the spec) this could be "somewhere" on `WebDriver` and not only on `RemoteWebDriver`. However, we try and keep the API of the main interfaces as compact as possible, and use role-based interfaces where necessary (which is why there are so many of them --- even for JS execution) 

Printing a page feels very similar to taking a screenshot, so we have two options to make this "feel" like a webdriver API. The first is to add a new method to `TakesScreenshot` (a default one that throws an exception, to keep our promise of Se4 being a drop-in replacement) The second is to add a new role-based interfaces (eg. `PrintsPage`) I guess a final option is to add a `print` method to `WebDriver.Window`, but that doesn't feel quite right.

Finally, the return type of this method tells us nothing. It's fine to use a [tiny type](https://darrenhobbs.com/2007/04/11/tiny-types/) for this kind of thing. In `TakesScreenshot` we use an `OutputType`. If we could munge things to and from PDF without additional libraries, I'd suggest adding `PDF` to `OutputType` and changing the print signature to be `print(OutputType<X> as)`, but that's not an option I'm entirely comfortable with as PDFBox adds about 4MB to the size of Selenium. As such, I'd recommend declaring a `Pdf` class and returning that.